From 6bbd0cf91e9539ef153d1e3ed120e4ca99a3b629 Mon Sep 17 00:00:00 2001
From: Yanjiang Jin <yanjiang.jin@windriver.com>
Date: Wed, 23 Mar 2016 13:33:17 +0800
Subject: [PATCH] ppc/e500mc: add book3e_die for ppc_e500mc

If E6500 CPU runs on 32bit mode, CPU wouldn't die during CPU offline
without this function.

Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 arch/powerpc/kernel/idle_e500.S |    8 ++++++++
 arch/powerpc/sysdev/fsl_rcpm.c  |    2 +-
 2 files changed, 9 insertions(+), 1 deletions(-)

diff --git a/arch/powerpc/kernel/idle_e500.S b/arch/powerpc/kernel/idle_e500.S
index 1544866..cbb3ea6 100644
--- a/arch/powerpc/kernel/idle_e500.S
+++ b/arch/powerpc/kernel/idle_e500.S
@@ -104,3 +104,11 @@ _GLOBAL(power_save_ppc32_restore)
 #endif
 
 	b	transfer_to_handler_cont
+
+#ifdef CONFIG_PPC_E500MC
+_GLOBAL(book3e_die)
+	wrteei  1
+	PPC_WAIT(0)
+1:
+	b	1b
+#endif /* E500MC */
diff --git a/arch/powerpc/sysdev/fsl_rcpm.c b/arch/powerpc/sysdev/fsl_rcpm.c
index d410b9a..e6b3f24 100644
--- a/arch/powerpc/sysdev/fsl_rcpm.c
+++ b/arch/powerpc/sysdev/fsl_rcpm.c
@@ -196,7 +196,7 @@ static void rcpm_v2_cpu_enter_state(int cpu, int state)
 		/* one bit corresponds to one thread for PH10 of 6500 */
 		setbits32(&rcpm_v2_regs->tph10setr0, 1 << hw_cpu);
 		break;
-#ifdef CONFIG_PPC_BOOK3E_64
+#ifdef CONFIG_PPC_E500MC
 	case E500_PM_PW10:
 		/* one bit corresponds to one thread for PW10 of 6500 */
 		book3e_die();
-- 
1.7.5.4

